lib/orbital/api.rb

```diff
@@ -228,17 +234,18 @@ def inquiry(order_id, trace_number, gateway)
         gateway.inquiry(order_id, trace_number)
       end
 
-      def retry_capture(plugin_trx_info, order_id, trace_number, context, gateway)
+      def retry_capture(plugin_trx_info, order_id, authorization, trace_number, context, gateway)
         options = {:trace_number => trace_number, :order_id => order_id}
-        kb_payment = @kb_apis.payment_api.get_payment(plugin_trx_info.kb_payment_id, false, false, [], context)
+
+        # Pass a cloned object of context to avoid being modified in get_payment via to_java
+        kb_payment = @kb_apis.payment_api.get_payment(plugin_trx_info.kb_payment_id, false, false, [], context.clone)
```

Comment:
FYI here's what we did for CyberSource: https://github.com/killbill/killbill-cybersource-plugin/blob/fdb48dad1f2ef3d8bc512de6e925ed9dceea9353/lib/cybersource/api.rb#L368-L369